src/kbmod/search/layered_image.cpp

```diff
@@ -17,6 +17,29 @@ LayeredImage::LayeredImage(const RawImage& sci, const RawImage& var, const RawIm
     science = sci;
     mask = msk;
     variance = var;
+
+    // Enforce consistency between layers.
+    validate_layers();
+}
+
+void LayeredImage::validate_layers() {
+    const int num_pixels = get_npixels();
+    float* sci_pixels = science.data();
+    float* var_pixels = variance.data();
+    float* mask_pixels = mask.data();
+
+    for (int i = 0; i < num_pixels; ++i) {
+        // Mask NaNs, infinities, or NO_DATA tags in sci or var layers consistently
+        // in all layers.
+        // Does not apply masking to the information to sci or var.
+        if (std::isnan(sci_pixels[i]) || std::isnan(var_pixels[i])
+            || (sci_pixels[i] == NO_DATA) || (var_pixels[i] == NO_DATA) 
+            || std::isinf(sci_pixels[i]) || std::isinf(var_pixels[i])) {
```

Comment:
I did not know that `NaN` is an actual thing in a standard, so these CPP `isnan` checks behave equally to python is-nan checks. I just assumed it's like a singleton defined somewhere in Numpy and they're not comparable. 

Would it be better for us then to use nan as the mask instead of `NO_DATA`? Do they exist on the GPU too? This could be an issue, doesn't have to be sorted in this PR.

Reply:
also, since there's an option in `reproject` to ignore bad values, it might make more sense to do it this way ^^